extensions/myext/testsrc/org/training/dao/impl/CustomProductDaoDefaultImplementationIntegrationTest.java

```diff
@@ -0,0 +1,81 @@
+package org.training.dao.impl;
+
+import de.hybris.platform.catalog.CatalogVersionService;
+import de.hybris.platform.catalog.enums.ArticleApprovalStatus;
+import de.hybris.platform.catalog.model.CatalogModel;
+import de.hybris.platform.catalog.model.CatalogVersionModel;
+import de.hybris.platform.core.model.product.ProductModel;
+import de.hybris.platform.servicelayer.ServicelayerTransactionalTest;
+import de.hybris.platform.servicelayer.exceptions.ModelNotFoundException;
+import de.hybris.platform.servicelayer.model.ModelService;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import javax.annotation.Resource;
+
+import static org.junit.Assert.*;
+
+public class CustomProductDaoDefaultImplementationIntegrationTest extends ServicelayerTransactionalTest {
+
+    private static final String PRODUCT_CODE = "121221212";
+    private static final String CATALOG_VERSION = "Staged";
+
+    private CatalogVersionModel catalogVersionModel;
+
+    @Resource(name = "customProductDao")
+    private CustomProductDaoDefaultImplementation productDao;
+
+    @Resource
+    private ModelService modelService;
+
+    @Resource
+    private CatalogVersionService catalogVersionService;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void setup() {
+        CatalogModel catalogModel = modelService.create(CatalogModel.class);
+        catalogModel.setId("100123131");
```

Comment:
It is better to extract all `String` values into constants. The same way as you did for `PRODUCT_CODE ` and `CATALOG_VERSION`